documentation/modules/auxiliary/scanner/msmq/cve_2023_21554_queuejumper.md

```diff
@@ -0,0 +1,56 @@
+[CVE-2023-21554](https://nvd.nist.gov/vuln/detail/CVE-2023-21554) ("QueueJumper") is a Remote Code Execution vulnerability with a CVSS 3.1 base score of 9.8 that could allow unauthenticated attackers to execute code on an unpatched Microsoft Windows system running [Microsoft Message Queuing (MSMQ)](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ms711472(v=vs.85)?redirectedfrom=MSDN).
+
+Security updates exist for Windows Server 2008 incl. R2, Windows Server 2012 incl. R2, Windows Server 2016, Windows Server 2019, Windows Server 2022, Windows 10 and Windows 11. MSMQ was first introduced with Windows NT 4.0 and Windows 2000, therefore it's likely that the vulnerability also exists and remains unpatched in unsupported Microsfot Windows versions.
+
+The module `auxiliary/scanner/msmq/cve_2023_21554_queuejumper` scans the given targets and detects whether a running instance of MSMQ is vulnerable to CVE-2032-21554. The module doesn't affect the stability of the MSMQ service, therefore it could be safely executed against the targets.
+
+## Vulnerable Application
+
+Microsoft Message Queuing (MSMQ) is a message queuing service that was first introduced with Windows NT 4.0 and exists in Microsoft Windows ever since. It needs to be explicitly installed, however many enterprise applications use MSMQ and also Microsoft Exchange installs MSMQ. Applications use MSMQ to send and retrieve messages from message queues.
+
+Besides several RPC-related TCP ports, MSMQ uses TCP port 1801 to receive messages from clients or other queue managers, leveraging the protocol [MS-MQQB](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mqqb/85498b96-f2c8-43b3-a108-c9d6269dc4af). By default all queues within a queue manager allow anonymous participants to send messages.
+
+## Verification Steps
```

Comment:
What would your thoughts be on adding a list of the know vulnerable targets here based of this [list](https://nvd.nist.gov/vuln/detail/CVE-2023-21554) of Known Affected Software Configurations. An [example](https://github.com/rapid7/metasploit-framework/blob/a276f336f09be7523d4b2211ee1276d52952c966/documentation/modules/exploit/windows/smb/ms17_010_eternalblue.md#vulnerable-application) from one of our other modules markdowns.

```suggestion
* Windows Server 2022
* Windows Server 2019
* ...
* ...

## Verification Steps
```

Reply:
Thank you. I think that'll be great for module users 👍